Maths/CollatzSequence.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * @function collatz
+ * @description Applies the Collatz Conjecture on a specified number.
+ * The Collatz Conjecture states that a number will always equal 1 after the following formula is applied:
+ * If the number is even, divide by 2, and if its odd, multiply it by 3 and add 1.
+ *
+ * @parama {Integer} n The number to apply the Collatz Conjecture to.
+ *
+ * @return An array of steps and the final result..
+ *
+ * @see [Collatz Conjecture](https://en.wikipedia.org/wiki/Collatz_conjecture)
+ *
+ * @example collatz(1) = { result: 1, steps: [] }
+ * @example collatz(5) = { result: 1, steps: [16, 8, 4, 2, 1] }
+*/
+export function collatz (n) {
+  const steps = []
+
+  while (n > 1) {
```

Comment:
Instead of `>=`, you should check if `n` is exactly equal to 1, which is more understandable. Also, you should check for all the 3 numbers, 1, 2, and 4, since you want to count the steps till `n` falls in the loop.

Reply:
No, it should be if n > 1, because the only way the Collatz Conjecture could be disproven is if the number would be less than one.

Reply:
Also if it were `n >= 1`, it would get into an infinite loop, because when you apply the Collatz Conjecture to 1, it will always be 1.

Reply:
> No, it should be if n > 1, because the only way the Collatz Conjecture could be disproven is if the number would be less than one.

That is actually wrong, because going to a number less than one following the Collatz rules is impossible. The number only decreases if it is even, and the smallest even number is 2, and 2/2 is 1. So no, the number will never be less than one.

The Collatz conjecture will be disproven if another cycle like the 1, 2, 4 cycle is found, or a number is found such that it keeps on increasing and never falls back down. Computationally, only the first is feasible, as the second is equivalent to the halting problem and is undecidable.

> Also if it were n >= 1, it would get into an infinite loop, because when you apply the Collatz Conjecture to 1, it will always be 1.

We want you to check if it is *not equal to 1*. So when it equals 1 the loop exits.

Reply:
Got it